tests/wol/conftest.py

```diff
@@ -11,3 +20,143 @@ def get_connected_dut_intf_to_ptf_index(duthost, tbinfo):
+    logging.info("Intf pair under vlan {}: {}".format(random_vlan, items_in_vlan))
+    return list(items_in_vlan)
+
+
+@pytest.fixture(scope="class")
+def random_intf_pair_to_remove_under_vlan(duthost, random_vlan, random_intf_pair):
+    duthost.del_member_from_vlan(vlan_n2i(random_vlan), random_intf_pair[0])
+    logging.info("Intf pair {} removed from vlan {}".format(random_intf_pair, random_vlan))
+
+    yield random_intf_pair
+
+    duthost.add_member_to_vlan(vlan_n2i(random_vlan), random_intf_pair[0], False)
+    logging.info("Intf pair {} added back to vlan {}".format(random_intf_pair, random_vlan))
+
+
+def setup_ip_on_ptf(duthost, ptfhost, ip, intf_pairs):
+    duthost.command('monit stop routeCheck', module_ignore_errors=True)
+    ptfhost.remove_ip_addresses()
+    ip = ipaddress.ip_address(ip)
+    if isinstance(ip, ipaddress.IPv4Address):
+        ping = "ping"
+    if isinstance(ip, ipaddress.IPv6Address):
+        ping = "ping6"
+    arp_responder_conf = {}
+    ping_commands = []
+    for dut_intf, ptf_index in intf_pairs:
+        arp_responder_conf["eth{}".format(ptf_index)] = [ip.__str__()]
+        ping_commands.append("{} -c 1 -w 1 -I {} {}".format(ping, dut_intf, ip))
+    with open(ARP_RESPONDER_PATH, "w") as f:
+        json.dump(arp_responder_conf, f)
+    ptfhost.copy(src=ARP_RESPONDER_PATH, dest=ARP_RESPONDER_PATH)
+    ptfhost.host.options["variable_manager"].extra_vars.update(
+            {"arp_responder_args": "--conf " + ARP_RESPONDER_PATH})
+    ptfhost.template(src="templates/arp_responder.conf.j2", dest="/etc/supervisor/conf.d/arp_responder.conf")
+    ptfhost.shell("supervisorctl reread && supervisorctl update")
+    ptfhost.shell("supervisorctl restart arp_responder")
+
+    duthost.command("sonic-clear fdb all")
+    duthost.command("sonic-clear arp")
+    duthost.command("sonic-clear ndp")
+    time.sleep(20)
+    for cmd in ping_commands:
+        duthost.shell(cmd, module_ignore_errors=True)
+
+
+def remove_ip_on_ptf(duthost, ptfhost):
+    ptfhost.shell("supervisorctl stop arp_responder")
+    ptfhost.shell("rm -f {}".format(ARP_RESPONDER_PATH))
+    ptfhost.shell("rm -f /etc/supervisor/conf.d/arp_responder.conf")
+    ptfhost.shell("supervisorctl reread && supervisorctl update")
+    duthost.command("sonic-clear fdb all")
+    duthost.command("sonic-clear arp")
+    duthost.command("sonic-clear ndp")
+    duthost.command('monit start routeCheck', module_ignore_errors=True)
+
+
+@pytest.fixture(scope="class")
+def dst_ip_intf(request, duthost, ptfhost, vlan_brief, random_vlan, random_intf_pair_to_remove_under_vlan):
+    ip = request.param
+    if ip == "ipv4" or ip == "ipv6":
+        vlan_intf = ipaddress.ip_interface(vlan_brief[random_vlan]["interface_" + ip][0])
+        ip = vlan_intf.network.broadcast_address.__str__()
+    if ip and isinstance(ipaddress.ip_address(ip), ipaddress.IPv6Address):
```

Comment:
Why we only setup ipv6? do we need ipv4 for this if-else also?

Reply:
ipv4 and ipv6 has some difference with broadcasting. With ipv6 its technically it has multicast not broadcast. Broadcast is sending to everyone regardless of sender and receiver. Multicast requires setting up valid ip on both sender and receiver.